pkg/supervisor/archived_snapshot.go

```diff
@@ -71,6 +75,12 @@ func (c *s3Snapshot) Upload(ctx context.Context, path string) error {
 		key = key[1:]
 	}
 	var reader io.Reader = bufio.NewReaderSize(f, 1024*32) // use a 32K buffer for reading
+
+	cs, err := getCheckSum(path)
```

Comment:
nit: Checksum not CheckSum. :)   (i.e., this is the noun of "[checksum](https://en.wikipedia.org/wiki/Checksum)")